# Датасеты для экстрактивной суммаризации

1. DebateSum: https://huggingface.co/datasets/Hellisotherpeople/DebateSum?row=0
- 241К строк, есть столбец с экстрактивной суммаризацией(текстом)
2. CNN-Dailymail: https://huggingface.co/datasets/eReverter/cnn_dailymail_extractive?row=7¶
- В общем 311К строк, есть столбец с маской преложений, которые относятся к экстрактивной суммаризации(В оригинальном датасете CNN только абстрактная суммаризация есть)
3. GovReport: https://huggingface.co/datasets/nianlong/long-doc-extractive-summarization-gov-report
- 19464 строк, надо разобраться что из себя представляют данные в столбце indices
4. Pubmed: https://huggingface.co/datasets/nianlong/long-doc-extractive-summarization-pubmed¶
- В общем 130К строк, тексты по медицинской тематике. ?Столбец indices
5. Arxiv: https://huggingface.co/datasets/nianlong/long-doc-extractive-summarization-arxiv
- Достаточно большой, train > 7Гб, загрузил пока только test и val. ?Столбец indices

(6) https://huggingface.co/datasets/dell-research-harvard/newswire - большой датасет из всяких новостных статей. Можно обучить на нём, а тестить, считать метрики на CNN, например.

In [8]:
import pandas as pd
from datasets import load_dataset

## 1. DebateSum - https://huggingface.co/datasets/Hellisotherpeople/DebateSum?row=0

In [2]:
DebateSum = load_dataset("Hellisotherpeople/DebateSum")
DebateSum

Extracting data files:   0%|          | 0/1 [00:00<?, ?it/s]

Generating train split: 0 examples [00:00, ? examples/s]

Dataset csv downloaded and prepared to C:/Users/Максат/.cache/huggingface/datasets/Hellisotherpeople___csv/Hellisotherpeople--DebateSum-d85e26ecd585ba24/0.0.0/6954658bab30a358235fa864b05cf819af0e179325c740e4bc853bcc7ec513e1. Subsequent calls will reuse this data.


  0%|          | 0/1 [00:00<?, ?it/s]

DatasetDict({
    train: Dataset({
        features: ['Unnamed: 0', 'Full-Document', 'Citation', 'Extract', 'Abstract', '#CharsDocument', '#CharsAbstract', '#CharsExtract', '#WordsDocument', '#WordsAbstract', '#WordsExtract', 'AbsCompressionRatio', 'ExtCompressionRatio', 'OriginalDebateFileName', 'DebateCamp', 'Tag', 'Year'],
        num_rows: 240566
    })
})

In [3]:
DebateSum.set_format(type="pandas")
debate_df = DebateSum['train'][:]
debate_df.head()

Unnamed: 0                                      Full-Document  \
0           0  How does one write the nation's modernity as t...   
1           1  Modernization theory provided post-war society...   
2           2  As we have seen in the previous chapter, the e...   
3           3  Frederick Jackson Turner (1962: 37-8), in his ...   
4           4  In the beginning colonialism was a product of ...   

                                            Citation  \
0  Bhabha 1994 (Homi K. Anne F. Rothenberg Profes...   
1  Slater 2004 (David, Professor of Social and Po...   
2  Slater 2004 (David, Professor of Social and Po...   
3  Slater 2004 (David, Professor of Social and Po...   
4  Quijano 2007 (Aníbal, PhD National University ...   

                                             Extract  \
0  why do nations celebrate their hoariness, not ...   
1  Modernization theory provided post-war society...   
2  , the emergence of the United States as a glob...   
3  the process of settlement and colonization bro...   
4  colonialism was a product of a systematic repr...   

                                            Abstract  #CharsDocument  \
0  A) The Affirmatives construction of a “Nation-...            1978   
1  Use of the term modernization is a blanket to ...            4400   
2  The projection of power portrayed of the US ga...            5047   
3  The AFF manifest in a coloniality of American ...            4285   
4  The assumed cultural norms of economic engagem...            1038   

   #CharsAbstract  #CharsExtract  #WordsDocument  #WordsAbstract  \
0             147            993             317              22   
1              65            459             689              11   
2             187           2273             784              31   
3             116           2860             675              18   
4             155            683             159              21   

   #WordsExtract  AbsCompressionRatio  ExtCompressionRatio  \
0            159             0.069401             0.501577   
1             67             0.015965             0.097242   
2            348             0.039541             0.443878   
3            445             0.026667             0.659259   
4            106             0.132075             0.666667   

                        OriginalDebateFileName     DebateCamp      Tag  Year  
0  Postcolonialism Kritik - Gonzaga 2013.html5  Gonzaga (GDI)  Kritiks  2013  
1  Postcolonialism Kritik - Gonzaga 2013.html5  Gonzaga (GDI)  Kritiks  2013  
2  Postcolonialism Kritik - Gonzaga 2013.html5  Gonzaga (GDI)  Kritiks  2013  
3  Postcolonialism Kritik - Gonzaga 2013.html5  Gonzaga (GDI)  Kritiks  2013  
4  Postcolonialism Kritik - Gonzaga 2013.html5  Gonzaga (GDI)  Kritiks  2013

In [4]:
drop_cols = debate_df.columns[(debate_df.columns!="Full-Document") &  (debate_df.columns!="Extract")]
drop_cols

Index(['Unnamed: 0', 'Citation', 'Abstract', '#CharsDocument',
       '#CharsAbstract', '#CharsExtract', '#WordsDocument', '#WordsAbstract',
       '#WordsExtract', 'AbsCompressionRatio', 'ExtCompressionRatio',
       'OriginalDebateFileName', 'DebateCamp', 'Tag', 'Year'],
      dtype='object')

In [5]:
for col in drop_cols:
    debate_df.pop(col)
debate_df.head(5)

Full-Document  \
0  How does one write the nation's modernity as t...   
1  Modernization theory provided post-war society...   
2  As we have seen in the previous chapter, the e...   
3  Frederick Jackson Turner (1962: 37-8), in his ...   
4  In the beginning colonialism was a product of ...   

                                             Extract  
0  why do nations celebrate their hoariness, not ...  
1  Modernization theory provided post-war society...  
2  , the emergence of the United States as a glob...  
3  the process of settlement and colonization bro...  
4  colonialism was a product of a systematic repr...

In [6]:
#debate_df.to_csv("DebateSum.csv")

## 2. CNN-Dailymail - https://huggingface.co/datasets/eReverter/cnn_dailymail_extractive?row=7

Вторая(первоначальная и более чистая) версия - https://huggingface.co/datasets/abisee/cnn_dailymail?row=0

In [7]:
CNN_Dailymail = load_dataset("eReverter/cnn_dailymail_extractive")
CNN_Dailymail

Extracting data files:   0%|          | 0/3 [00:00<?, ?it/s]

Generating train split:   0%|          | 0/287113 [00:00<?, ? examples/s]

Generating validation split:   0%|          | 0/13368 [00:00<?, ? examples/s]

Generating test split:   0%|          | 0/11490 [00:00<?, ? examples/s]

Dataset parquet downloaded and prepared to C:/Users/Максат/.cache/huggingface/datasets/eReverter___parquet/eReverter--cnn_dailymail_extractive-acc148585d3f84c8/0.0.0/2a3b91fbd88a2c90d1dbbb32b460cf621d31bd5b05b934492fdef7d8d6f236ec. Subsequent calls will reuse this data.


  0%|          | 0/3 [00:00<?, ?it/s]

DatasetDict({
    train: Dataset({
        features: ['src', 'tgt', 'labels'],
        num_rows: 287113
    })
    validation: Dataset({
        features: ['src', 'tgt', 'labels'],
        num_rows: 13368
    })
    test: Dataset({
        features: ['src', 'tgt', 'labels'],
        num_rows: 11490
    })
})

In [8]:
CNN_Dailymail.set_format(type='pandas')

CNN_train_df = CNN_Dailymail['train'][:]
CNN_val_df = CNN_Dailymail['validation'][:]
CNN_test_df = CNN_Dailymail['test'][:]

CNN_train_df.head(5)

src  \
0  [LONDON, England (Reuters) -- Harry Potter sta...   
1  [Editor's note: In our Behind the Scenes serie...   
2  [MINNEAPOLIS, Minnesota (CNN) -- Drivers who w...   
3  [WASHINGTON (CNN) -- Doctors removed five smal...   
4  [(CNN)  -- The National Football League has in...   

                                                 tgt  \
0  [Harry Potter star Daniel Radcliffe gets £20M ...   
1  [Mentally ill inmates in Miami are housed on t...   
2  [NEW: "I thought I was going to die," driver s...   
3  [Five small polyps found during procedure; "no...   
4  [NEW: NFL chief, Atlanta Falcons owner critica...   

                                              labels  
0  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...  
1  [0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  
2  [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  
3  [1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  
4  [1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...

Описание датасета:
- `src`: исходный текст
- `tgt`: абстрактная суммаризация текста
- `labels`: бинарная маска предложений, из которых составляется экстрактивная суммаризация

По идее нам будут полезны только столбцы src и labels, но пока оставлю все как есть

In [10]:
#CNN_train_df.to_csv("CNN-2/CNN-train.csv")
#CNN_val_df.to_csv("CNN-2/CNN-val.csv")
#CNN_test_df.to_csv("CNN-2/CNN-test.csv")

Следующие датасеты не выходит получить через load_dataset, а через API можно получить только первые несколько строк. Пришлось скачивать json-ы.

## 3. GovReport - https://huggingface.co/datasets/nianlong/long-doc-extractive-summarization-gov-report

In [7]:
gov_train_df = pd.read_json("train.jsonl", lines=True)
gov_val_df = pd.read_json("val.jsonl", lines=True)
gov_test_df = pd.read_json("test.jsonl", lines=True)
gov_train_df.head()

text  \
0  [The rental assistance programs authorized und...   
1  [In its 2005 opinion United States v. Booker ,...   
2  [The Randolph-Sheppard Act, originally signed ...   
3  [C ongress has long sought to ensure that alie...   
4  [The 111 th Congress is considering legislatio...   

                                             summary  \
0  [The Section 8 low-income housing program is r...   
1  [Pursuant to the Anti-Drug Abuse Act of 1986, ...   
2  [The Randolph-Sheppard Act requires that blind...   
3  [The Immigration and Nationality Act (INA) has...   
4  [Legislation to augment telework in executive ...   

                                             indices  \
0  [[9, 14, 7, 0, 119, 85, 32, 69, 2, 3, 101, 78,...   
1  [[164, 3, 255, 0, 153, 238, 123, 144, 124, 234...   
2  [[38, 42, 5, 43, 6, 0], [38, 42, 5, 4, 43, 0],...   
3  [[73, 8, 13, 6, 105, 19, 17, 56, 26, 113, 80, ...   
4  [[86, 100, 78, 87, 103, 79, 91, 76, 101, 89, 1...   

                                               score  
0  [0.575668231839745, 0.5756186547845571, 0.5759...  
1  [0.5911478200672481, 0.5910184409533751, 0.591...  
2  [0.5108167770419421, 0.514526862522317, 0.5054...  
3  [0.665898224397371, 0.665363360719874, 0.66489...  
4  [0.879655545240251, 0.8793091955764271, 0.8793...

In [8]:
gov_train_df.to_csv("GovReport-train.csv")

In [11]:
gov_val_df.to_csv("GovReport-val.csv")

In [12]:
gov_test_df.to_csv("GovReport-test.csv")

## 4. Pubmed - https://huggingface.co/datasets/nianlong/long-doc-extractive-summarization-pubmed

In [2]:
pubmed_train_df = pd.read_json("train.jsonl", lines=True)
pubmed_val_df = pd.read_json("val.jsonl", lines=True)
pubmed_test_df = pd.read_json("test.jsonl", lines=True)
pubmed_train_df.head()

text  \
0  [a recent systematic analysis showed that in 2...   
1  [it occurs in more than 50% of patients and ma...   
2  [tardive dystonia ( td ) , a rarer side effect...   
3  [lepidoptera include agricultural pests that ,...   
4  [syncope is caused by transient diffuse cerebr...   

                                             summary  \
0  [background : the present study was carried ou...   
1  [backgroundanemia in patients with cancer who ...   
2  [tardive dystonia ( td ) is a serious side eff...   
3  [many lepidopteran insects are agricultural pe...   
4  [we present an unusual case of recurrent cough...   

                                             indices  \
0  [[115, 108, 152, 114, 120, 118], [115, 108, 15...   
1  [[52, 53, 9, 72, 40, 35, 3], [52, 53, 9, 72, 2...   
2  [[28, 0, 26, 27, 3, 2], [28, 0, 2, 26, 27], [2...   
3  [[0, 31, 23, 170, 13], [169, 31, 23, 0, 13], [...   
4  [[4, 8, 3, 5, 28, 18, 37], [4, 8, 34, 5, 18, 3...   

                                               score  
0  [0.5892528524107471, 0.585056226698186, 0.5789...  
1  [0.5943213570658511, 0.5943213570658511, 0.585...  
2  [0.414947056087242, 0.41404033433018905, 0.386...  
3  [0.398238269666841, 0.38964992389649905, 0.387...  
4  [0.46287187529702506, 0.46045548654244306, 0.4...

In [3]:
pubmed_train_df.to_csv("Pubmed-train.csv")

In [4]:
pubmed_val_df.to_csv("Pubmed-val.csv")

In [5]:
pubmed_test_df.to_csv("Pubmed-test.csv")

## 5. Arxiv - https://huggingface.co/datasets/nianlong/long-doc-extractive-summarization-arxiv

Скачал только val и test части датасета.

In [21]:
arxiv_val_df = pd.read_json("val.jsonl", lines=True)
arxiv_val_df.head()

text  \
0  [the interest in anchoring phenomena and pheno...   
1  [galaxy clusters , as the largest peaks in the...   
2  [quantum correlations between components of a ...   
3  [methanol masers are often found in star - for...   
4  [interdisciplinary research has recently gaine...   

                                             summary  \
0  [we study the phase behavior of a nematic liqu...   
1  [determining the scaling relations between gal...   
2  [we show how to control spatial quantum correl...   
3  [class  i methanol masers are believed to be p...   
4  [nowadays , scientific challenges usually requ...   

                                      sorted_indices  
0  [233, 22, 84, 228, 92, 192, 27, 43, 20, 65, 18...  
1  [578, 29, 136, 583, 30, 99, 594, 590, 508, 85,...  
2  [26, 15, 32, 138, 3, 18, 38, 4, 136, 146, 13, ...  
3  [157, 73, 141, 154, 86, 42, 156, 57, 115, 135,...  
4  [95, 74, 8, 7, 23, 26, 93, 78, 6, 101, 88, 5, ...

In [22]:
arxiv_val_df.to_csv("Arxiv/Arxiv-val.csv")

In [23]:
arxiv_test_df = pd.read_json("test.jsonl", lines=True)
arxiv_test_df.head()

text  \
0  [for about 20 years the problem of properties ...   
1  [it is believed that the direct detection of g...   
2  [as a common quantum phenomenon , the tunnelin...   
3  [for the hybrid monte carlo algorithm ( hmc)@x...   
4  [recently it was discovered that feynman integ...   

                                             summary  \
0  [the short - term periodicities of the daily s...   
1  [we study the detectability of circular polari...   
2  [starting from the wkb approximation , a new b...   
3  [we study a novel class of numerical integrato...   
4  [new methods for obtaining functional equation...   

                                      sorted_indices  
0  [218, 217, 31, 33, 195, 216, 161, 203, 6, 125,...  
1  [149, 6, 147, 146, 19, 36, 62, 14, 98, 16, 144...  
2  [79, 12, 10, 84, 22, 13, 23, 73, 25, 52, 77, 8...  
3  [10, 18, 8, 72, 14, 27, 85, 0, 54, 96, 82, 90,...  
4  [163, 132, 0, 88, 4, 5, 17, 21, 148, 141, 161,...

In [24]:
arxiv_test_df.to_csv("Arxiv/Arxiv-test.csv")

## Newswire - большой датасет, https://huggingface.co/datasets/dell-research-harvard/newswire
Возьмем пока что первые 115К строк.

In [2]:
import requests
import json

In [6]:
response = requests.get("https://huggingface.co/api/datasets/dell-research-harvard/newswire/parquet/default/train/0.parquet")
response

<Response [200]>

In [11]:
with open("newswire.parquet", 'wb') as file:
    file.write(response.content)

In [12]:
result = pd.read_parquet("newswire.parquet")
result.head(5)

ca_topic wire_state  protests  civil_rights  \
0                         Health                    0             0   
1  Federal Government Operations   new york         0             0   
2                          Fires  minnesota         0             0   
3          Churches and Religion                    0             0   
4                    Agriculture                    0             0   

  wire_location_notes                                   people_mentioned  \
0                                                                     []   
1                                                                     []   
2                      [{'person_gender': 'man', 'person_name': 'Cadw...   
3                                                                     []   
4                                                                     []   

                                  newspaper_metadata  \
0  [{'lccn': 'sn89066350', 'newspaper_city': 'chi...   
1  [{'lccn': 'sn84023963', 'newspaper_city': 'jas...   
2  [{'lccn': 'sn85033078', 'newspaper_city': 'gra...   
3  [{'lccn': 'sn87090488', 'newspaper_city': 'vic...   
4  [{'lccn': 'sn82014805', 'newspaper_city': 'wil...   

                                          ner_labels       wire_coordinates  \
0  [O, O, O, O, O, O, O, O, O, O, O, O, O, O, O, ...                     []   
1  [O, O, O, B-ORG, B-LOC, I-LOC, O, B-PER, O, O,...  [40.71427, -74.00597]   
2  [B-LOC, B-LOC, O, B-PER, O, O, O, O, O, O, O, ...  [44.97997, -93.26384]   
3  [B-LOC, O, B-PER, O, O, I-MISC, O, O, O, O, B-...   [41.89193, 12.51133]   
4  [O, O, B-ORG, B-LOC, I-ORG, O, O, O, O, O, O, ...                     []   

   crime  ...  antitrust                                          ner_words  \
0      0  ...          0  [upon, our, country, We, are, at, peace, with,...   
1      0  ...          0  [Special, to, the, Star, NEW, York, Oct, 1, TH...   
2      0  ...          0  [MNSEAPoLIs, MIMI, May, 2, At, z, o, clock, th...   
3      0  ...          0  [RoNE, March, 2, s, A, Consistory, was, held, ...   
4      0  ...          0  [From, the, Carrolton, Mo, Democrat, The, last...   

                               byline    wire_city  year govt_regulation  \
0                                                   1878               0   
1                                         New York  1878               0   
2                                      Minneapolis  1878               0   
3                                             Rome  1878               0   
4  [From the Carrolton (Mo.) Democrat               1878               0   

                                             article  \
0  upon our country.\n\n\nWe are at peace with al...   
1  Special to the Star.\n\n\nNEW York, Oct. 1.-TH...   
2  MNSEAPoLIs, MIMI, May 2.-At z\no'clock this ev...   
3  RoNE, March 2s.--A Consistory was held\nat the...   
4  From the Carrolton (Mo.) Democrat\n\n\nThe las...   

                                               dates cluster_size  \
0  [Dec-19-1878, Dec-04-1878, Dec-02-1878, Dec-05...           22   
1  [Nov-15-1878, Oct-25-1878, Oct-31-1878, Oct-24...            8   
2  [May-03-1878, May-04-1878, May-09-1878, May-15...           12   
3            [Mar-29-1878, Mar-28-1878, Apr-05-1878]            8   
4  [Jul-30-1878, Sep-13-1878, Aug-01-1878, Nov-07...           11   

    wire_country  
0                 
1  United States  
2  United States  
3          Italy  
4                 

[5 rows x 22 columns]

In [13]:
len(result)

115198

Удалим лишние столбцы

In [14]:
result = result["article"]
result.head()

0    upon our country.\n\n\nWe are at peace with al...
1    Special to the Star.\n\n\nNEW York, Oct. 1.-TH...
2    MNSEAPoLIs, MIMI, May 2.-At z\no'clock this ev...
3    RoNE, March 2s.--A Consistory was held\nat the...
4    From the Carrolton (Mo.) Democrat\n\n\nThe las...
Name: article, dtype: object

In [16]:
result = pd.DataFrame(result)
result

article
0       upon our country.\n\n\nWe are at peace with al...
1       Special to the Star.\n\n\nNEW York, Oct. 1.-TH...
2       MNSEAPoLIs, MIMI, May 2.-At z\no'clock this ev...
3       RoNE, March 2s.--A Consistory was held\nat the...
4       From the Carrolton (Mo.) Democrat\n\n\nThe las...
...                                                   ...
115193  washington, Dec 2s-President\nCleveland this a...
115194  ST. JOSEPH, Mo-, Dec. 21.-John F.\nKunz, the s...
115195  Sioux City Iowa. Dec. 2s-Donald\nMclean. prote...
115196  Leadville, Col, Dec. 26.-About 2 o'clock\nthis...
115197  LONDON, Dec. 26.-The Times Paris\ncorresponden...

[115198 rows x 1 columns]

In [17]:
#result.to_csv("Newswire/newswire-0.csv")

In [20]:
print(result.loc[0]['article'])

upon our country.


We are at peace with all other nations
Our public credit has greatly improved, ant
is, perhaps, now stronger than ever before
Abundant haryests have rewarded the labor.
of those who till the soil, our manufacturing
industries are reyiying, and it is believed
that general prosperity, which has been SC
anxiously looked for, is at'least within oul
reach.


THE YELLOW FEVER.


The enjoyment of health by our people
generally has, however, been interrupted,
during the past season, by the prevalence OF
a fatal pestilence, the yellow fever, in some
portions of the Southern States, creating an
emergency which called for prompt and ex-
traordinary measures Of relief. The disease
appeared as an epidemic at New Orleans and
at other places on the lower Mississippi, soon
after midsummer. 1t was rapidly spread
ny fugitiyes from the infected cities and
towns, and did not disappear till early in No
yember. The States of Louisiana, Mississ-
ippi and Tennessee, have suffered severely.